src/shared/directives/conditional-show/services/condition-parser.service.ts

```diff
@@ -0,0 +1,202 @@
+          return expectedValues === values;
+        }
+    }
+
+    private static checkValueContains(expectedValue: string, currentValue: any): boolean {
+        if (expectedValue.search(',') > -1) {
+            let expectedValues = expectedValue.split(',').sort();
+            let values = currentValue ? currentValue.sort().toString() : '';
+            return expectedValues.every(item => values.search(item) >= 0);
+        } else {
+            let values = currentValue && Array.isArray(currentValue) ? currentValue.toString() : '';
+            return values.search(expectedValue) >= 0;
+        }
+    }
+
+    private static unquoted(str: string): string {
+        const res = str.replace(/^"|"$/g, '');
+        return res;
+    }
+
+    private static findValueForComplexCondition(fields: object, head: string, tail: string[], path?: string): any {
+        if (!fields) {
+          return undefined;
+        }
+        if (tail.length === 0) {
+          return this.getValue(fields, head);
+        } else {
+          if (FieldsUtils.isArray(fields[head])) {
+            return this.findValueForComplexConditionInArray(fields, head, tail, path);
+          } else {
+            return this.findValueForComplexConditionForPathIfAny(fields, head, tail, path);
+          }
+        }
+    }
+
+    private static findValueForComplexConditionForPathIfAny(fields: object, head: string, tail: string[], path?: string): any {
+        if (path) {
+            const [_, ...pathTail] = path.split(/[_]+/g);
+            return this.findValueForComplexCondition(fields[head], tail[0], tail.slice(1), pathTail.join('_'));
+        } else {
+            return this.findValueForComplexCondition(fields[head], tail[0], tail.slice(1), path);
+        }
+    }
+
+    private static findValueForComplexConditionInArray(fields: object, head: string, tail: string[], path?: string): any {
+    // use the path to resolve which array element we refer to
+        if (path.startsWith(head)) {
+            const [_, ...pathTail] = path.split(/[_]+/g);
+            if (pathTail.length > 0) {
+                try {
+                    let arrayIndex = Number.parseInt(pathTail[0], 10);
+                    const [__, ...dropNumberPath] = pathTail;
+                    return (fields[head][arrayIndex] !== undefined) ? this.findValueForComplexCondition(
+                    fields[head][arrayIndex]['value'], tail[0], tail.slice(1), dropNumberPath.join('_')) : null;
+                } catch (e) {
+                    console.log('Error while parsing number', pathTail[0], e);
+                }
+            }
+        } else {
+            console.log('Path in formArray should start with ', head, ', full path: ', path);
+        }
+    }
+
+    private static removeStarChar(str: string): string {
+        return str.substring(0, str.length - 1);
+    }
+
+    private static okIfBothEmpty(right: string, value: any): boolean {
+        return value === null && (right === '');
```

Comment:
What about if `value === undefined` or `value === ''` or maybe even `value === []`? 🤔

Reply:
this is old code so keeping it as it is.